test/spmd/test_fsdp_v2.py

```diff
@@ -0,0 +1,45 @@
+import unittest
+import os
+import sys
+
+import torch
+import torch_xla
+import torch_xla.runtime as xr
+import torch_xla.core.xla_model as xm
+import torch_xla.distributed.spmd as xs
+
+import test_xla_sharding_base
+import torch_xla.experimental.spmd_fully_sharded_data_parallel as fsdp
+
+
+class BasicShardingTest(test_xla_sharding_base.XlaShardingTest):
+
+  @classmethod
+  def setUpClass(cls):
+    xr.use_spmd()
+    super().setUpClass()
+
+  def test_fsdp_v2(self):
+    model = self.SimpleLinear().to(xm.xla_device())
+    mesh = self._get_mesh((self.n_devices, 1), None, ('fsdp', 'tensor'))
+    model.fc1 = fsdp.SpmdFullyShardedDataParallel(model.fc1, mesh)
```

Comment:
Do we need to shard the individual layers? Since the full model is wrapped on L27

Reply:
I would also go a step further to test 2 cases
- wrap one of the inners and then the outer, check if all are wrapped; let's say the two wrapping uses different shardings and make sure the original sharding is unchanged.
- wrap the outer, and try and except when wrapping the inner

Reply:
I hope this use case is well explained in the design doc.